cpu/cc26xx_cc13xx/power_arch.c

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (C) 2020 Locha Inc
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_cc26xx_cc13xx
+ * @{
+ *
+ * @file
+ * @brief       Power management abstractions
+ *
+ * @author      Jean Pierre Dudey <jeandudey@hotmail.com>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+
+#define DOMAIN_ON (1)
+
+/* Save changes of the PRCM */
+static void prcm_commit(void)
+{
+    /* Write CLKLOADCTL in the non-buffered register bank to avoid buffered
+     * writes */
+    PRCM_NONBUF->CLKLOADCTL = CLKLOADCTL_LOAD;
+
+    /* Wait while load is done */
+    while (!(PRCM->CLKLOADCTL & CLKLOADCTL_LOADDONE)) {}
+}
+
+bool power_is_domain_enabled(const power_domain_t domain)
+{
+    switch (domain) {
+        case POWER_DOMAIN_PERIPHERALS:
+            return PRCM->PDSTAT0PERIPH == DOMAIN_ON;
+
+        case POWER_DOMAIN_SERIAL:
+            return PRCM->PDSTAT0SERIAL == DOMAIN_ON;
+
+        case POWER_DOMAIN_RFC:
+            return (PRCM->PDSTAT1RFC == DOMAIN_ON) &&
+                   (PRCM->PDSTAT0RFC == DOMAIN_ON);
+
+        case POWER_DOMAIN_VIMS:
+            return PRCM->PDSTAT1VIMS == DOMAIN_ON;
+
+        case POWER_DOMAIN_CPU:
+            return PRCM->PDSTAT1CPU == DOMAIN_ON;
+
+        default:
+            return false;
+    }
+
+    return false;
+}
+
+void power_enable_domain(const power_domain_t domain)
+{
+    switch (domain) {
+        case POWER_DOMAIN_PERIPHERALS:
+            PRCM->PDCTL0PERIPH = DOMAIN_ON;
+            break;
+
+        case POWER_DOMAIN_SERIAL:
+            PRCM->PDCTL0SERIAL = DOMAIN_ON;
+            break;
+
+        case POWER_DOMAIN_RFC:
+            PRCM->PDCTL0RFC = DOMAIN_ON;
+            PRCM->PDCTL1RFC = DOMAIN_ON;
+            break;
+
+        case POWER_DOMAIN_CPU:
+            PRCM->PDCTL1CPU = DOMAIN_ON;
+            break;
+
+        case POWER_DOMAIN_VIMS:
+            PRCM->PDCTL1VIMS = DOMAIN_ON;
+            break;
+    }
+
+    while (!power_is_domain_enabled(domain)) {}
+}
+
+void power_clock_enable_gpio(void)
+{
+    PRCM->GPIOCLKGR = GPIOCLKGR_CLK_EN;
+
+    prcm_commit();
+}
+
+void power_clock_enable_gpt(uint32_t tim)
+{
+    PRCM->GPTCLKGR |= (1 << tim);
+
+    prcm_commit();
+}
+void power_clock_enable_i2c(void) {
+    PRCM->I2CCLKGR = I2CCLKGR_CLK_EN;
+
+    prcm_commit();
+}
+
+void power_clock_enable_uart(uint32_t uart)
+{
+    if (uart == 0) {
+        PRCM->UARTCLKGR |= UARTCLKGR_CLK_EN_UART0;
+    }
+#ifdef CPU_VARIANT_X2
+    else if (uart == 1) {
+        PRCM->UARTCLKGR |= UARTCLKGR_CLK_EN_UART1;
+    }
+#endif
+
+    prcm_commit();
+}
+
+void power_clock_disable_uart(uint32_t uart)
+{
+    if (uart == 0) {
+        PRCM->UARTCLKGR &= ~UARTCLKGR_CLK_EN_UART0;
+    }
+#ifdef CPU_VARIANT_X2
```

Comment:
Would this also work?
```suggestion
#ifdef UARTCLKGR_CLK_EN_UART1
```